File: Data Structures & Algorithms/maximum-subarray/submission-6.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        maxsum = nums[0]
        n = len(nums)
        if n <= 1:
            return nums[0]

        current = nums[0]
        i = 1
        while i < n:
            if current < 0:
                current = nums[i]
            else:
                current += nums[i]
            maxsum = max(maxsum, current)
            i += 1

        return maxsum

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-6.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int res = 0 , i = 0,j=i;
        int n = s.size();
        unordered_map<char,int> mp;
        while(j < n){
            if(mp[s[j]] > 0){
               while(i<j && s[i]!=s[j]){
                 mp[s[i]]--;
                 i++;
               }
               mp[s[i]]--;
               i++;
            }
            res = max(res, j-i+1);
            mp[s[j]]++;
            j++;
        }
        return res;
    }
};